integration-test-groups/foundation/variables/src/main/java/org/apache/camel/quarkus/variables/it/VariablesResource.java

```diff
@@ -0,0 +1,195 @@
+
+        MockEndpoint mid = context.getEndpoint("mock:removeLocalVariableMid", MockEndpoint.class);
+        mid.expectedMessageCount(1);
+
+        producerTemplate.requestBody("direct:removeLocalVariableStart", body, String.class);
+
+        // make sure we got the message
+        end.assertIsSatisfied();
+        mid.assertIsSatisfied();
+
+        // lets get the variable value
+        List<Exchange> exchanges = end.getExchanges();
+        Exchange endExchange = exchanges.get(0);
+        exchanges = mid.getExchanges();
+        Exchange midExchange = exchanges.get(0);
+
+        return midExchange.getVariable(VariablesRoutes.VARIABLE_NAME, String.class) + ","
+                + endExchange.getVariable(VariablesRoutes.VARIABLE_NAME, String.class);
+    }
+
+    @Path("/removeGlobalVariable")
+    @POST
+    public String removeGlobalVariable(String body) throws Exception {
+        MockEndpoint mid = context.getEndpoint("mock:removeGlobalVariableMid", MockEndpoint.class);
+        mid.expectedMessageCount(1);
+
+        producerTemplate.requestBody("direct:removeGlobalVariableStart", body, String.class);
+
+        // make sure we got the message
+        mid.assertIsSatisfied();
+
+        // lets get the variable value
+        List<Exchange> exchanges = mid.getExchanges();
+        Exchange midExchange = exchanges.get(0);
+
+        return midExchange.getVariable(VariablesRoutes.VARIABLE_NAME, String.class) + ","
+                + context.getVariable(VariablesRoutes.VARIABLE_NAME, String.class);
+    }
+
+    @Path("/customGlobalRepository")
+    @POST
+    public Response customGlobalRepository(String body) throws Exception {
+        //        context.getRegistry().bind(GLOBAL_VARIABLE_REPOSITORY_ID, new MyGlobalRepo());
+
+        context.getRouteController().startRoute("customGlobalRepository");
+
+        MockEndpoint end = context.getEndpoint("mock:setGlobalCustomEnd", MockEndpoint.class);
+        end.expectedMessageCount(1);
+
+        producerTemplate.requestBody("direct:setGlobalCustomStart", body, String.class);
+
+        // make sure we got the message
+        end.assertIsSatisfied();
+
+        // lets get the variable value
+        List<Exchange> exchanges = end.getExchanges();
+        Exchange exchange = exchanges.get(0);
+
+        String resp = exchange.getVariable(VariablesRoutes.VARIABLE_NAME, String.class) + ","
+                + context.getVariable(VariablesRoutes.VARIABLE_NAME, String.class);
+        return Response.ok().entity(resp).build();
+    }
+
+    @Path("/convert")
+    @POST
+    public String convert(String body) throws Exception {
+        MockEndpoint end = context.getEndpoint("mock:convertEnd", MockEndpoint.class);
+        end.expectedMessageCount(1);
+
+        producerTemplate.requestBody("direct:convertStart", body, String.class);
+        //        fluentProducerTemplate.to("direct:convertStart").withVariable(VariablesRoutes.VARIABLE_NAME, body).send();
```

Comment:
Same here, could we remove that ?

Reply:
It seems, that I reverted more code then anticipated, because I was removing that a while ago